common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -868,8 +872,23 @@ class @Problem
       next_index = parseInt(hint_index) + 1
     $.postWithPrefix "#{@url}/hint_button", hint_index: next_index, input_id: @id, (response) =>
       if response.success
-        @render(response.html, @focus_on_hint_notification)
-        hint_container = @.$('.problem-hint')
+        hint_msg_container = @.$('.problem-hint .notification-message')
         hint_container.attr('hint_index', response.hint_index)
+        hint_msg_container.html(response.msg)
+        # Update any Mathjax entries
+        MathJax.Hub.Queue [
```

Comment:
I don't know. It seems to be working, but that may be because the MathJax processing is asychronous (so it is still processing after it has been made visible). Elsewhere in the file, it appears to me that MathJax is being called after the HTML is visible.

There are bok choy tests for this though, and if they are passing, then theoretically we are covered.